EcalTiming/plugins/EcalTimingCalibProducer.cc

```diff
@@ -389,17 +391,6 @@ EcalTimingCalibProducer::Status EcalTimingCalibProducer::endOfLoop(const edm::Ev
 	dumpCorrections(filename);
 	// save the xml
 
-
-	timeEBRing.dumpToTree(dumpTree,   EBRING,  0, -2 , DS_EB_RING);
-	timeEEmRing.dumpToTree(dumpTree, EEmRING, 0, -10, DS_EEm_RING);
-	timeEEpRing.dumpToTree(dumpTree, EEpRING, 0, 10 , DS_EEp_RING);
-
-	EBDetId eb(EBCRYex);
-	EEDetId ee(EECRYex);
-	timeEBCRYex.dumpToTree(dumpTree, eb.ieta(), eb.iphi(), 0,        DS_EB_CRYS);
-	timeEECRYex.dumpToTree(dumpTree, ee.ix(),   ee.iy(), ee.zside(), DS_EE_CRYS);
```

Comment:
They way it was coded, a crystal could end up in the tree more than once if
it was being dumped for a different reason as well. Now i check if the
crystal fits the requirements of these rings/rawid at the time of dump and
just dump in then, instead of storing all of them and dumping at the end.

See
https://github.com/previsualconsent/EcalTiming/blob/master/EcalTiming/plugins/EcalTimingCalibProducer.cc#L348
through line 358

On Thu, Jul 2, 2015 at 4:38 AM, shervin86 <notifications@github.com> wrote:

> In EcalTiming/plugins/EcalTimingCalibProducer.cc
> <https://github.com/ECALELFS/EcalTiming/pull/6#discussion_r33761553>:
>
> > @@ -389,17 +391,6 @@ EcalTimingCalibProducer::Status EcalTimingCalibProducer::endOfLoop(const edm::Ev
> >  	dumpCorrections(filename);
> >  	// save the xml
> >
> > -
> > -	timeEBRing.dumpToTree(dumpTree,   EBRING,  0, -2 , DS_EB_RING);
> > -	timeEEmRing.dumpToTree(dumpTree, EEmRING, 0, -10, DS_EEm_RING);
> > -	timeEEpRing.dumpToTree(dumpTree, EEpRING, 0, 10 , DS_EEp_RING);
> > -
> > -	EBDetId eb(EBCRYex);
> > -	EEDetId ee(EECRYex);
> > -	timeEBCRYex.dumpToTree(dumpTree, eb.ieta(), eb.iphi(), 0,        DS_EB_CRYS);
> > -	timeEECRYex.dumpToTree(dumpTree, ee.ix(),   ee.iy(), ee.zside(), DS_EE_CRYS);
>
> Why have you removed it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ECALELFS/EcalTiming/pull/6/files#r33761553>.
>